core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedRegionPlanGenerator.scala

```diff
@@ -194,55 +204,74 @@ class CostBasedRegionPlanGenerator(
     )
 
     while (queue.nonEmpty) {
+      // A state is represented as a set of materialized non-blocking edges.
       val currentState = queue.dequeue()
       visited.add(currentState)
+
       tryConnectRegionDAG(
         physicalPlan.getNonMaterializedBlockingAndDependeeLinks ++ currentState
       ) match {
         case Left(regionDAG) =>
-          // Calculate the current state's cost and update the bestResult if it's lower
-          val cost =
-            evaluate(regionDAG.vertexSet().asScala.toSet, regionDAG.edgeSet().asScala.toSet)
-          if (cost < bestResult.cost) {
-            bestResult = SearchResult(currentState, regionDAG, cost)
-          }
-        // No need to explore further
+          checkSchedulableState(regionDAG)
+          expandFrontier()
```

Comment:
I know this is from bfs terminology. however, can we give the function name a bit more context?

Reply:
done